pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -60,20 +74,26 @@ func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *an
 		return controller.Result{}, err
 	}
 
-	return controller.NewPhaseRunner().Register(
-		r.ipValidator.ValidateControlPlaneIP,
+	return controller.NewPhaseRunner[*Scope]().Register(
+		r.ValidateControlPlaneIP,
 		r.ValidateClusterSpec,
 		r.ValidateHardware,
 		r.ValidateDatacenterConfig,
 		r.ReconcileControlPlane,
 		r.CheckControlPlaneReady,
 		r.ReconcileCNI,
 		r.ReconcileWorkers,
-	).Run(ctx, log, clusterSpec)
+	).Run(ctx, log, NewScope(clusterSpec))
```

Comment:
We should init a new scope before phase register so the values in the scope can be updated thru phases. This way `isRollingUpgrade` will be reset to false